File: ResoniteCacheCleanUp.js
const fs = require('fs');
const path = require('path');
const os = require('os');

const limit = new Date();
limit.setDate(limit.getDate() - 7);

const userProfile = os.homedir();
const path1 = path.join(userProfile, 'AppData', 'LocalLow', 'Yellow Dog Man Studios', 'Resonite', 'Assets');
const path2 = path.join(userProfile, 'AppData', 'Local', 'Temp', 'Yellow Dog Man Studios', 'Resonite', 'Cache');

cleanFolder(path1);
cleanFolder(path2);

function cleanFolder(folderPath) {
  if (fs.existsSync(folderPath)) {
    fs.readdirSync(folderPath).forEach((file) => {
      const filePath = path.join(folderPath, file);
      const stats = fs.statSync(filePath);
      if (stats.atime < limit) {
        if (stats.isFile()) {
          fs.unlinkSync(filePath);
          console.log(`Deleted file: ${filePath}`);
        } else if (stats.isDirectory()) {
          fs.rmdirSync(filePath, { recursive: true });
          console.log(`Deleted directory and its contents: ${filePath}`);
        }
      }
    });
  }
}

File: ResoniteCacheCleanUp.py
import os
import datetime

# Get the user's folder (e.g., C:\Users\drago)
user_folder = os.path.expanduser("~")

def delete_files_older_than_7_days(directory):
    limit = datetime.datetime.now() - datetime.timedelta(days=7)
    
    for root, dirs, files in os.walk(directory):
        for file in files:
            file_path = os.path.join(root, file)
            file_stat = os.stat(file_path)
            file_mtime = datetime.datetime.fromtimestamp(file_stat.st_mtime)
            
            if file_mtime < limit:
                os.remove(file_path)
                print(f"Deleted: {file_path}")

directories = [
    os.path.join(user_folder, "AppData", "LocalLow", "yellow Dog Man Studios", "Resonite", "Assets"),
    os.path.join(user_folder, "AppData", "Local", "Temp", "Yellow Dog Man Studios", "Resonite", "Cache")
]

for directory in directories:
    delete_files_older_than_7_days(directory)
